### PR TITLE
Updated brain.py, hand.py and game_state.py

### PR DESCRIPTION
**Pull request by Froopie**

### CODE REVIEW

File: brain.py [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
+        self.type = type
+        self.p = probability
+
+    def set_type(self, type):
+        self.type = type
     
-    # User's brain
-    def response0(self, state):
+    def set_probability(self, p):
+        self.p = p
```
Comment (NouemanKHAL) [inline brain.py:18]: check if  `0 <= p <= 1` otherwise throw an exception or set some default value and print a warning

File: brain.py [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
 
 class Brain:
-    def __init__(self, _type):
-        self.type = _type
+    def __init__(self, type=0, probability=0.5):
+        self.type = type
+        self.p = probability
+
+    def set_type(self, type):
+        self.type = type
```
Comment (NouemanKHAL) [inline brain.py:15]: check if type is a valid type in case there is a list of allowed types

File: card.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 from constants import *
 class Card:
-    def __init__(self, _type, value):
-        self.type = _type
+    def __init__(self, type, value):
```
Comment (NouemanKHAL) [inline card.py:3]: well, it works, but here's the reason I don't recommend using `type` https://stackoverflow.com/a/10568115

File: hand.py [hunk 4]
```diff
@@ -13,6 +13,14 @@ def has(self, value):
                 return True
         return False
 
+    def cards_of_value(self, value):
```
Comment (NouemanKHAL) [inline hand.py:16]: ambiguous function name

File: card.py (hunk 3 above)
Comment (Froopie) [reply card.py]: Changed type to suit everywhere

File: hand.py (hunk 4 above)
Comment (Froopie) [reply hand.py:17]: Added a comment

File: brain.py (hunk 2 above)
Comment (Froopie) [reply brain.py:15]: Will be done once all bots are established

**Pull request merged**